File: primerTrimestre/introduccion_programacion_python/3repetitivas_py/ej14rep.py
# Programa: ej14rep.py
# Propósito: Realizar un programa que compruebe si una cadena contiene una subcadena. 
# Las dos cadenas se introducen por teclado.
#
# Autor: Antonio Quesada
# Fecha: 30/10/2019.
#
# Variables a usar:
#   * cadena
#   * subcadena
# subcadena = cadena[i:i+len(cadena)]:


#Petición de datos.
cadena = input("Introduce una cadena: ")
subcadena = input("Introduce de nuevo otra cadena: ")

#Bucle for con un rango de 1.
for i in range(0,1):
    #Funcion contains para ver si la cadena contiene a la subcadena.
    if(cadena.__contains__(subcadena)):
        print("La cadena contiene la subcadena.")
        
    #En el caso de que no la contenga lo imprime por pantalla.
    else:
        print("La cadena no contiene la subcadena.")
        

File: primerTrimestre/java basico/3repetitivas/Ej14rep.java
/**Realizar un programa que compruebe si una cadena contiene una subcadena. Las dos cadenas se
 * introducen por teclado.
 *
 * @author Antonio Quesada
 */

import java.util.Scanner;

public class Ej14rep {
  public static void main(String[] args) {
    Scanner s = new Scanner(System.in);
    
    //Leemos una cadena y subcadena.
    System.out.println("Introduce una cadena: ");
    String cadena = s.nextLine();
    System.out.println("Introduce una cadena: ");
    String subcadena = s.nextLine();
    
    //Mientras que la cadena contenga (contains())a la subcadena lo muestra. 
  while(cadena.contains(subcadena)) {
    System.out.println("La cadena contiene la subcadena.");
    break;
  }
       //Si la cadena no contiene a la subcadena.
    if(!cadena.contains(subcadena)) {
      System.out.println("La cadena no contiene a la subcadena");
     
    }
    } 
 
  }


